### PR TITLE
Add a lambda handler for Benchmark.AI Reporting.

### PR DESCRIPTION
**Pull request by vishaalkapoor**

### CODE REVIEW

File: reporting/report.py [hunk 1]
```diff
@@ -82,3 +84,26 @@
     if args.email_addr:
         report_html = io.open(args.report_file + HTML_EXTENSION, mode='r', encoding='utf-8').read()
         email_report(report_html, args.email_addr)
+
+
+def lambda_handler(event, context):
```
Comment (sad-) [inline reporting/report.py]: would it be better to have the lambda_handler in its own file? unless the logic in` __name__ = '__main__' ` block is tightly coupled to the lambda handler and the intention is to have all the parse arguments and stuff run first?

Comment (vishaalkapoor) [reply reporting/report.py]: Sure - better sooner than later as over time the report.py file will get cluttered. I'll split it up.

File: reporting/scripts/update_lambda.sh [hunk 2]
```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+# Updates the Lambda code in AWS to generate the report daily.
+# Ensure AWS environment variables 
+# The AWS CLI requires the following environment variables to be set accordingly.
+# * AWS_ACCESS_KEY_ID
+# * AWS_SECRET_ACCESS_KEY
+# * AWS_DEFAULT_REGION
+# Run this from the main directory, e.g. scripts/update_lambda.sh
+
+set -e
```
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply reporting/scripts/update_lambda.sh]: Done

File: reporting/report.py [hunk 3]
```diff
@@ -82,3 +84,26 @@
...
+    """
+    Main entry point for AWS Lambda.
+    The EMAIL environment variable is required to be set.
+    """
+    logging.info("Reading configuration and fetching metrics from Cloudwatch.")
+    benchmarks = Benchmarks()
+
+    logging.info('Generating report.')
+    REPORT_FILE = '/tmp/bai-report'
+    generate_report(REPORT_FILE, benchmarks)
```
[the remark this answers is not in the document]
Comment (vishaalkapoor) [reply reporting/report.py]: I'm following the failfast pattern using exceptions. generate_report will throw if any of the file operations fail.

**Comment by vishaalkapoor** [comment]
Thanks for reviews.
@sad-  Please take a look. Thanks!

**Comment by vishaalkapoor** [comment]
Thanks!

**Pull request merged**